dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerifyUtility.java

```diff
@@ -0,0 +1,182 @@
+            }
+            String errorMsg = errorMsgBuild.toString();
+            throw new ValidateYamlException(INVALID_FILE_STRUCTURE + errorMsg);
+        }
+    }
+
+    public static void dockstoreValidate(String path) throws ValidateYamlException {
+        Path workflowPath = Paths.get(path);
+
+        // Verify that the path is a valid directory
+        if (!Files.isDirectory(workflowPath)) {
+            throw new ValidateYamlException(INVALID_DOCKSTORE_YML + workflowPath.toString() + FILE_DOES_NOT_EXIST);
+        }
+
+        // Verify that .dockstore.yml exists
+        Path dockstoreYmlPath = Paths.get(path, DOCKSTOREYML);
+        if (!Files.exists(dockstoreYmlPath)) {
+            throw new ValidateYamlException(INVALID_DOCKSTORE_YML + dockstoreYmlPath.toString() + FILE_DOES_NOT_EXIST);
+        }
+
+        // Verify that .dockstore.yml is non-empty
+        File dockstoreYml = new File(dockstoreYmlPath.toString());
+        if (dockstoreYml.length() == 0) {
+            throw new ValidateYamlException(INVALID_DOCKSTORE_YML + dockstoreYmlPath.toString() + EMPTY_FILE);
+        }
+
+        // Load dockstoreYml into string
+        String dockstoreYmlString = "";
+        try {
+            dockstoreYmlString = Files.readString(dockstoreYmlPath);
+        } catch (IOException ex) {
+            // Unlikely to ever catch
+            throw new ValidateYamlException("Error reading " + dockstoreYmlPath.toString() + " :\n" + ex.getMessage());
+        }
+
+        // Verify that the Yaml is valid, however does not verify it is valid for dockstore
+        Yaml yaml = new Yaml();
+        try {
+            yaml.load(Files.readString(dockstoreYmlPath));
+            out(dockstoreYmlPath + VALID_YAML_ONLY);
+        } catch (Exception ex) {
+            throw new ValidateYamlException(INVALID_DOCKSTORE_YML + dockstoreYmlPath.toString() + INVALID_YAML + ex.getMessage());
+        }
+
+        // Running validate first, as if readAsDockstoreYaml12 is run first and a parameter is incorrect it is difficult to understand
+        // Determine if any of the parameters names are incorrect (does not verify if all the parameters are there)
+        try {
+            DockstoreYamlHelper.validateDockstoreYamlProperties(dockstoreYmlString);
+        } catch (Exception ex) {
+            throw new ValidateYamlException(INVALID_DOCKSTORE_YML + dockstoreYmlPath.toString() + CONTAINS_ERRORS + ex.getMessage());
+        }
+
+        // Validates file dockstoreYaml in general
+        DockstoreYaml12 dockstoreYaml12 = null;
+        try {
+            dockstoreYaml12 = DockstoreYamlHelper.readAsDockstoreYaml12(dockstoreYmlString);
+        } catch (Exception ex) {
+            throw new ValidateYamlException(INVALID_DOCKSTORE_YML + dockstoreYmlPath.toString() + CONTAINS_ERRORS + ex.getMessage());
+        }
+
+        // Determines if all referenced files exist
+        allFilesExist(dockstoreYaml12, workflowPath.toString());
```

Comment:
I have added support for `.github` directories. Now `YamlVerifyUtility` will check if it is in a `.github` directory and if it is it will check all of the paths in `.dockstore.yml` relative to its parent directory.